be/src/olap/delete_handler.cpp

```diff
@@ -110,7 +110,7 @@ std::string DeleteHandler::construct_sub_predicates(const TCondition& condition)
         } else if (op == "!*=") {
             op = "!=";
         }
-        condition_str = condition.column_name + op + condition.condition_values[0];
+        condition_str = condition.column_name + op + "'" + condition.condition_values[0] + "'";
```

Comment:
If we upgrade from doris 1.2 and there is a delete condition. It does not contains '', is there error?

Reply:
because it already save the previous condition does not have '' in data file.